platform/ios/src/MGLMapView.mm

```diff
@@ -1721,7 +1722,18 @@ - (void)handlePanGesture:(UIPanGestureRecognizer *)pan
 
         if ([self _shouldChangeFromCamera:oldCamera toCamera:toCamera])
         {
-            self.mbglMap.moveBy({ delta.x, delta.y });
+
+            switch(self.scrollingMode){
+               case MGLScrollingModeVertical:
+                  self.mbglMap.moveBy({ 0, delta.y });
+                  break;
+               case MGLUserScrollingModeHorizontal:
+                  self.mbglMap.moveBy({ delta.x, 0 });
+                  break;
+               default:
+                  self.mbglMap.moveBy({ delta.x, delta.y });
+            }
+
```

Comment:
If you limited scrolling, then when the user scrolls in a perpendicular direction you still get some movement which feels a bit odd. 

What do you think about scaling the amount the map moves by, by a factor that takes this into account?

Reply:
Per chat we'll defer this for a follow-up PR if needed.